src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -45,12 +45,14 @@ pub struct WorkUnit {
 #[derive(Clone, Default)]
 pub struct WorkUnitStore {
   workunits: Arc<Mutex<HashSet<WorkUnit>>>,
+  new_workunits_since_last_check: Arc<Mutex<HashSet<WorkUnit>>>,
```

Comment:
Uhm. I'm not a fan of managing extra state and keeping potentially twice as much memory for what is conceptually a sliding window.

HashSets don't play amazingly with views as they're not contiguous in memory, but how would you feel about:
* refactoring workunits into using a `Vec` instead of a `HahSet` (at a glance, I don't see a compelling reason to use HashSet here, but we would need to confirm that with the callers of `get_workunits`)
* new_workunits_since_last_check being replaced by an `Option<usize>` being the index of the last reported workunit if any workunit is present
* get_latest_workunits returning a slice of the workunits?

Reply:
I implemented basically the schema you suggested